src/goto-symex/goto_state.cpp

```diff
@@ -31,19 +31,17 @@ std::size_t goto_statet::increase_generation(
   const ssa_exprt &lhs,
   std::function<std::size_t(const irep_idt &)> fresh_l2_name_provider)
 {
-  std::size_t n = fresh_l2_name_provider(l1_identifier);
+  const std::size_t n = fresh_l2_name_provider(l1_identifier);
 
-  const auto r_opt = level2.current_names.find(l1_identifier);
-
-  if(!r_opt)
+  if(const auto r_opt = level2.current_names.find(l1_identifier))
   {
-    level2.current_names.insert(l1_identifier, std::make_pair(lhs, n));
+    std::pair<ssa_exprt, unsigned> copy = r_opt->get();
+    copy.second = narrow<unsigned>(n);
```

Comment:
Ack, but we should really stop using `unsigned`, the renaming index ought to be `std::size_t`.

Reply:
@tautschnig I added that change in a new commit

Reply:
Thank you!